buildkite/resource_cluster.go

```diff
@@ -75,25 +78,50 @@ func (c *clusterResource) Schema(ctx context.Context, req resource.SchemaRequest
 				MarkdownDescription: "A color representation of the Cluster. Accepts hex codes, eg #BADA55.",
 			},
 		},
+		Blocks: map[string]resource_schema.Block{
+			"timeouts": timeouts.Block(ctx, timeouts.Opts{
+				Create: true,
+			}),
+		},
 	}
 }
 
 func (c *clusterResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
 	var state *clusterResourceModel
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
+
+	diags := req.Plan.Get(ctx, &state)
+
+	resp.Diagnostics.Append(diags...)
 
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	r, err := createCluster(
-		c.client.genqlient,
-		c.client.organizationId,
-		state.Name.ValueString(),
-		state.Description.ValueStringPointer(),
-		state.Emoji.ValueStringPointer(),
-		state.Color.ValueStringPointer(),
-	)
+	createTimeout, diags := state.Timeouts.Create(ctx, DefaultTimeout)
+
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var r *createClusterResponse
+	err := retry.RetryContext(ctx, createTimeout, func() *retry.RetryError {
+		var err error
+		r, err = createCluster(
+			c.client.genqlient,
+			c.client.organizationId,
+			state.Name.ValueString(),
+			state.Description.ValueStringPointer(),
+			state.Emoji.ValueStringPointer(),
+			state.Color.ValueStringPointer(),
+		)
+		if err != nil {
+			return retry.NonRetryableError(err)
+		}
+
+		return nil
+	})
```

Comment:
**Summary of my code diving:**  
I think this all makes sense. `createTimeout` is the maximum time the client will retry, which includes N retries within that time until it can get a value. The SDK has baked in to wait 500ms between tries up to 10s max. So it'll run through a few retries by the time it hits `DefaultTimeout`.

This should capture non-2xx response codes from the API. I'm thinking 401/3 which we don't want to retry, but we do want to allow retrying on 429 and possibly some/most 5xx codes. This is not captured currently so won't really resolve #214.

Graphql is also pretty tricky in that it returns 200 status codes but an `error` field in the response JSON. This is not captured by this either.

**Some things to figure out:**  
- Do we want to have a maximum timeout users can't go above? I think maybe not, at least at first. The timeout block is the absolute timeout which includes retrying so its not the timeout of the HTTP requests themselves
- Per above, we could extend this to cancel the HTTP requests themselves by making use of https://github.com/Khan/genqlient/blob/main/docs/genqlient.yaml#L55-L61 which would be separate from this timeout. I think the `timeout` block per resource makes sense to be a maximum timeout for that resource to be operated on by terraform. A HTTP level timeout would be less than that
- How do we figure out the status code or graphql error that occurred? So that we can make use of the `retry.RetryableError`

Reply:
Turns out genqlient does it for us already: https://github.com/Khan/genqlient/blob/main/graphql/client.go#L161-L177

So if there is a HTTP level error, we get that returned. 
If there is a json decode error (ie wrong object) we get that returned.
If there are errors in the json response, we get that returned.

The only thing that is missing is access to response headers for things like rate-limiting. But through the `retry.Retry` we have no place to change the wait between retries anyway to make use of the specified time